google-cloud-pubsublite/src/main/java/com/google/cloud/pubsublite/internal/wire/PublisherImpl.java

```diff
@@ -236,6 +237,13 @@ void flushToStream() {
     }
   }
 
+  @VisibleForTesting
```

Comment:
Please remove this method. You can check the permanent error handlerr to determine if a permanent error occurred.

Reply:
Yeah, that was the first thing I tried, but the listeners don't get called. I traced through and think it's because [AbstractService::notifyFailed](https://github.com/google/guava/blob/master/android/guava/src/com/google/common/util/concurrent/AbstractService.java#L451) just throws an IllegalStateException when the state is New or Terminated, and doesn't transition to Failed. Since there's no state transition, the listeners aren't called.

Reply:
Put it into STARTING or STOPPING by making the startAsync or stopAsync method of the RetryingConnection block on a CountdownLatch. Then try to publish to trigger this case, and then let the method proceed.

Reply:
Thanks, CountDownLatch is useful. I looked into your suggestion and noticed that all of our doStarts are synchronous. The only scenario that really triggers the original bug (b/156525288) is when the user publishes before calling startAsync, and so left the test to check NEW.

I removed the isShutdown() and just verified that the service has failed permanently by checking that an IllegalStateException is thrown if the user then tries to start the service. This occurs if they try to start PublisherImpl or any proxy such as WrappingPublisher.